src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,375 @@
+﻿using System.Collections.Generic;
+using Godot;
+using Godot.Collections;
+using static CustomDialog;
+
+/// <summary>
+///   Reorders windows by reordering their ancestors that are on the same level as this node.
+/// </summary>
+/// <remarks>
+///   <para>
+///     The windows are the ones who ask to establish the connection. If the window asks for a recursive
+///     connection then the connection gets passed to window reordering nodes in ancestors too.
+///   </para>
+///   <para>
+///     IMPORTANT: Don't attach this class to any node, always use the existing scene with the same name as this class.
+///     The reason for this is because the name of the node is used when establishing connections.
+///   </para>
+/// </remarks>
+public class AddWindowReorderingSupportToSiblings : Control
+{
+    /// <summary>
+    ///   Paths to window reordering nodes in ancestors.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     This overrides automatic search.
+    ///   </para>
+    /// </remarks>
+    [Export]
+    public Array<NodePath> WindowReorderingPaths = new();
+
+    /// <summary>
+    ///   Tries to finds first window reordering node in ancestors to connect to
+    ///   up to the specified depth.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Ignored when window reordering paths are not empty.
+    ///   </para>
+    /// </remarks>
+    [Export]
+    public int AutomaticWindowReorderingDepth = 5;
+
+#pragma warning disable CA2213
+
+    /// <summary>
+    ///   Pairs containing window reordering nodes and what sibling of theirs is an ancestor of this class.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Used to pass the connections from windows.
+    ///   </para>
+    /// </remarks>
+    private List<(AddWindowReorderingSupportToSiblings ReorderingNode, Node Sibling)> windowReorderingAncestors =
```

Comment:
Now as this is a list, this doesn't need to be inside the pragma, and I think this should be now able to be changed to a readonly. Otherwise this variable will break the rule that all Godot node types should be before other variables of a class.